libs/base/src/utils/CConfigFileMemory_unittest.cpp

```diff
@@ -0,0 +1,60 @@
+/* +---------------------------------------------------------------------------+
+   |                     Mobile Robot Programming Toolkit (MRPT)               |
+   |                          http://www.mrpt.org/                             |
+   |                                                                           |
+   | Copyright (c) 2005-2017, Individual contributors, see AUTHORS file        |
+   | See: http://www.mrpt.org/Authors - All rights reserved.                   |
+   | Released under BSD License. See details in http://www.mrpt.org/License    |
+   +---------------------------------------------------------------------------+ */
+
+#include <mrpt/utils/CConfigFileMemory.h>
+#include <mrpt/system/os.h>
+
+#include "simpleini/SimpleIni.h"
+
+#include <gtest/gtest.h>
+
+using namespace mrpt;
+using namespace mrpt::utils;
+using namespace mrpt::utils::simpleini;
+using namespace std;
+
+class TConfig : public CConfigFileMemory {
+	public:
+	using CConfigFileMemory::writeString;
+	using CConfigFileMemory::readString;
+	using CConfigFileMemory::getAllSections;
+	using CConfigFileMemory::getAllKeys;
+	};
```

Comment:
ah, I see!

I just added [this commit](https://github.com/MRPT/mrpt/commit/b6dd211cffd73ec09af01e54321d2e87aadd6f1d) which hopefully will solve your confusion. Those writeString() & readString() are *virtual* methods, not intended for usage from the user's side. 

getAllSections() and getAllKeys() are public, you should have no problem directly using them...